docs - update public links (#27022)
metabase/metabase: docs/questions/sharing/public-links.md

---
title: Public sharing
redirect_from:
  - /docs/latest/administration-guide/12-public-links
  - /docs/latest/embedding/12-public-links
  - /docs/latest/questions/sharing/public-links
---

# Public sharing

Sometimes you'll want to share a dashboard or question you've saved with someone that isn't a part of your organization or company, or someone who doesn't need access to your full Metabase instance. Metabase lets administrators create public links and [public embeds](#public-embed) to let you do just that.

## Enable public sharing on Metabase

![Enable public sharing](../images/enable-public-sharing.png)

> An admin needs to enable public sharing from **Settings** > **Admin settings** > **Public sharing** before you'll see the [sharing icon](#enable-sharing-on-your-saved-question-or-dashboard) option on questions and dashboards.

Reviewer: ```suggestion
In order to share Metabase items like questions and dashboards via a public link or an embedded iframe, an admin needs to enable public sharing for that Metabase instance by going to **Settings gear icon** > **Admin settings** > **Public sharing**.
```

Once toggled on, the **Public sharing** section will display Metabase dashboards and questions with active public links. To make a public link inactive, click **Revoke link**.

## Enable sharing on your saved question or dashboard

> If [public sharing](#enable-sharing-on-your-saved-question-or-dashboard) is enabled for your Metabase, you'll find the `Sharing and Embedding` icon on saved questions and dashboards (it looks like an arrow pointing up and to the right). The sharing button is located on the bottom right corner of a question, or the top right corner of a dashboard.

Reviewer: ```suggestion
If [public sharing](#enable-sharing-on-your-saved-question-or-dashboard) is enabled for your Metabase, you'll find the **Sharing and Embedding** icon on saved questions and dashboards (it looks like an arrow pointing up and to the right). The sharing button is located on the bottom right corner of a question, or the top right corner of a dashboard.
```

> Click on the sharing icon to bring up the sharing settings modal, then click the toggle to generate the sharing link for that question or dashboard.

Reviewer: ```suggestion
To generate the sharing link for that question or dashboard, click on the sharing icon to bring up the sharing settings modal, then click the toggle.
```

We should state the action first, so people can decide to keep reading or bail if the action is irrelevant to their goal.

If you want to embed the link to your dashboard or question in a simple web page or blog post, copy and paste the iframe snippet to your destination of choice.

> You can update the link in the `src` attribute with [public embed parameters](#public-embed-parameters) to customize the appearance of your question or dashboard.

Reviewer: ```suggestion
To customize the appearance of your question or dashboard, you can update the link in the `src` attribute with [public embed parameters](#public-embed-parameters). 
```

## Public embed parameters

> You can add hash parameters at the end of the public link in your iframe's `src` attribute to toggle appearance settings or set filter values. 

Reviewer: ```suggestion
To toggle appearance settings or set filter values, you can add hash parameters at the end of the public link in your iframe's `src` attribute.
```

> Note that it's possible to find the a public link URL behind a public embed. If someone gets access to the public link URL, they can remove the hash parameters from the URL to view the original question or dashboard (without any appearance or filter settings).

Reviewer: Good note here.

If you'd like to create a secure embed that prevents people from changing filter names or values, check out [signed embedding](../../embedding/signed-embedding.md).

### Appearance

| Parameter name         | Possible values                                  |
| ---------------------- | ------------------------------------------------ |
| bordered               | true, false                                      |
| titled                 | true, false                                      |
| theme                  | null, transparent, night                         |

> | font                   | [font name](../../configuring-metabase/fonts.md) |

Reviewer: Fonts are paid, and though it might be worth mentioning here, we can probably rely on the link to communicate that.
